src/org/mozilla/javascript/SecurityBridge_SecurityManager.java

```diff
@@ -0,0 +1,103 @@
+/* -*- Mode: java; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.javascript;
+
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.AllPermission;
+import java.security.Permission;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+
+/**
+ * Code moved from {@link SecurityUtilities}. This implementation makes use of {@link
+ * SecurityManager} and {@link AccessController} and so on, which is deprecated with JDK17 (see <a
+ * href='https://openjdk.java.net/jeps/411'>JEP411</a>) - so all related 'java.security' stuff
+ * should be routed over this class, so that it could be easily replaced by an other implementation
+ * like {@link SecurityBridge_NoOp}.
+ *
+ * <p>This implementation should be work up to JDK17
```

Comment:
According to JEP 411, the only API that will not work out the box in Java 18 will be `System.setSecurityManager()` (to make it work, one needs to set `java -Djava.security.manager=allow`). Rhino only calls `System.setSecurityManager()` in `SecurityControllerTest`, so this implementation should work in Java 18 as well. It is unclear when the actual APIs are going to be removed.

I'd suggest setting `java.security.manager=allow` in the tests section of `build.gradle` (or I could do that in a separate PR) to deal with `SecurityControllerTest`, and avoid mentioning which Java release will/won't work.